scripts/coccinelle/semantic_parentheses.cocci

```diff
@@ -0,0 +1,13 @@
+// Copyright (C) 2019 Network RADIUS SAS.  Licenced under CC-by-NC 4.0.
+// Development of this scripts was sponsored by Network RADIUS SAS.
+// Author: Jorge Pereira (jpereira@freeradius.org)
+// Confidence: High
+// Comments: Fix semantic parentheses
+// Options: --no-includes --include-headers
+
+@@
+expression E;
+constant C;
+@@
+- !E & C
++ !(E & C)
```

Comment:
Hmm, useful, but wouldn't this change the meaning of a negated expression, or does '!' mean something different here? Part of the point of this ticket was to make Coccinelle easier for other developers to use, so these examples really need to be documented better.

Reply:
![image](https://user-images.githubusercontent.com/254448/72632451-12390c80-3935-11ea-9c3f-af3d40335268.png)

from http://coccinelle.lip6.fr/papers/tutorial.pdf page titled `A semantic patch for the !& problem`. But, I could remove it.